library/testing/TestSDKEngineRecreation.cxx

```diff
@@ -0,0 +1,20 @@
+#include <engine.h>
+#include <interactor.h>
+
+#include <iostream>
+
+int TestSDKEngineRecreation(int argc, char* argv[])
+{
+  std::string filename = "TestSDKEngineRecreation";
+  std::string interactionFilePath = std::string(argv[1]) + "/recordings/" + filename + ".log";
+  f3d::engine* eng = new f3d::engine(f3d::engine::create(false));
+  eng->getInteractor().start(1 / 30, [eng]() { eng->getInteractor().stop(); });
+  delete eng;
+  for (int i = 0; i < 5; i++)
+  {
+    f3d::engine* eng2 = new f3d::engine(f3d::engine::create(false));
+    eng2->getInteractor().playInteraction(interactionFilePath);
+    delete eng2;
+  }
```

Comment:
let's modernize the memory management using unique_ptr's
```suggestion
  auto eng = std::make_unique<f3d::engine>(f3d::engine::create(false));
  eng->getInteractor().start(1 / 30, [eng]() { eng->getInteractor().stop(); });
  eng = nullptr;
  for (int i = 0; i < 5; i++)
  {
    auto eng2 = std::make_unique<f3d::engine>(f3d::engine::create(false));
    eng2->getInteractor().playInteraction(interactionFilePath);
  }
```

Reply:
can you scope the first engine too instead of doing `= nullptr`?

Reply:
it's possible too. I'm fine with both